internal/resolver/delegatingresolver/delegatingresolver.go

```diff
@@ -0,0 +1,195 @@
+		r.isProxy = false
+	}
+	if !r.isProxy {
+		return targetResolverBuilder.Build(target, cc, opts)
+	}
+	if target.URL.Scheme == "dns" {
+		r.targetAddrs = []resolver.Address{{Addr: r.target.Endpoint()}}
+	} else {
+		r.targetResolver, err = targetURIResolver(target, opts, targetResolverBuilder, r)
+	}
+	r.ProxyResolver, err = proxyURIResolver(r.proxyURL, opts, r)
+	if err != nil {
+		return nil, err
+	}
+	return r, nil
+}
+
+func targetURIResolver(target resolver.Target, opts resolver.BuildOptions, targetResolverBuilder resolver.Builder, resolver *delegatingResolver) (resolver.Resolver, error) {
+	targetBuilder := targetResolverBuilder
+	if targetBuilder == nil {
+		return nil, fmt.Errorf("resolver for target scheme %q not found", target.URL.Scheme)
+	}
+	return targetBuilder.Build(target, &innerClientConn{resolver, "target"}, opts)
+}
+
+func proxyURIResolver(proxyURL *url.URL, opts resolver.BuildOptions, presolver *delegatingResolver) (resolver.Resolver, error) {
+	scheme := "dns"
+	proxyBuilder := resolver.Get(scheme)
+	if proxyBuilder == nil {
+		return nil, fmt.Errorf("resolver for proxy not found")
+	}
+	host := "dns:///" + proxyURL.Host
+	u, err := url.Parse(host)
+	if err != nil {
+		return nil, err
+	}
+
+	proxyTarget := resolver.Target{URL: *u}
+	return proxyBuilder.Build(proxyTarget, &innerClientConn{presolver, "proxy"}, opts)
+}
+
+func (r *delegatingResolver) ResolveNow(o resolver.ResolveNowOptions) {
+	if r.targetResolver != nil {
+		r.targetResolver.ResolveNow(o)
+	}
+	if r.ProxyResolver != nil {
+		r.ProxyResolver.ResolveNow(o)
+	}
+}
+
+func (r *delegatingResolver) Close() {
+	if r.targetResolver != nil {
+		r.targetResolver.Close()
+	}
+	if r.ProxyResolver != nil {
+		r.ProxyResolver.Close()
+	}
+}
+
+// UpdateState intercepts state updates from the childtarget and proxy resolvers.
+func (icc *innerClientConn) UpdateState(state resolver.State) error {
+	icc.mu.Lock()
+	defer icc.mu.Unlock()
+
+	var curState resolver.State
+	if icc.resolverType == "target" {
+		icc.targetAddrs = state.Addresses
+		curState = state
+	}
+	if icc.resolverType == "proxy" {
+		icc.proxyAddrs = state.Addresses
+	}
+
+	if len(icc.targetAddrs) == 0 || len(icc.proxyAddrs) == 0 {
+		return nil
+	}
+
+	var addresses []resolver.Address
+	for _, proxyAddr := range icc.proxyAddrs {
+		for _, targetAddr := range icc.targetAddrs {
+			attr := attributes.New("proxyConnectAddr", targetAddr.Addr)
+			if icc.proxyURL.User != nil {
+				attr = attr.WithValue("user", icc.proxyURL.User)
+			}
+			addresses = append(addresses, resolver.Address{
+				Addr:       proxyAddr.Addr,
+				Attributes: attr,
+			})
+		}
+	}
+	// Set the addresses in the current state.
+	curState.Addresses = addresses
+	return icc.cc.UpdateState(curState)
+}
+
+// ReportError intercepts errors from the child DNS resolver.
+func (icc *innerClientConn) ReportError(err error) {
+	icc.cc.ReportError(err)
+}
+
+func (icc *innerClientConn) NewAddress(addrs []resolver.Address) {
```

Comment:
Is this a no-op by design? if so, why?

Reply:
I made it a no op because it is deprecated.